encoder.go

```diff
@@ -42,16 +42,54 @@ func putEncoder(enc *logfmtEncoder) {
 	_logfmtPool.Put(enc)
 }
 
+// Custom logger encoding configuration.
+type config struct {
+	alternativeEncodeCaller zapcore.CallerEncoder
```

Comment:
nit
```suggestion
	alternativeCallerEncoder zapcore.CallerEncoder
```

Reply:
good point, also that suggested change 😄